CODING_STANDARDS.md

```diff
@@ -0,0 +1,378 @@
+- Imports should also be ordered by types (non-type-only imports first) then depth and then alphabetically (case-insensitive). Exceptions are:
+    - if a 3rd party library requires being the first import
+    - if within a React file `import React from 'react';` should always be first
+    - if within a React Native file `import {} from 'react-native';` goes after the previous exception before all other alphabetical imports
+    
+_Example:_
+
+```tsx
+    import React, { PropsWithChildren, useEffect, useState } from 'react';
+    import { Platform } from 'react-native';
+    import Dayjs from 'dayjs';
+
+    import { useCreateChatContext } from './hooks/useCreateChatContext';
+    import { useIsOnline } from './hooks/useIsOnline';
+
+    import {
+        ChatContextValue,
+        ChatProvider,
+    } from '../../contexts/chatContext/ChatContext';
+    import { useOverlayContext } from '../../contexts/overlayContext/OverlayContext';
+    import {
+        DeepPartial,
+        ThemeProvider,
+    } from '../../contexts/themeContext/ThemeContext';
+    import {
+        TranslationContextValue,
+        TranslationProvider,
+    } from '../../contexts/translationContext/TranslationContext';
+    import { useStreami18n } from '../../utils/useStreami18n';
+
+    import type { Channel } from 'stream-chat';
+
+    import type { Theme } from '../../contexts/themeContext/utils/theme';
+    import type { Streami18n } from '../../utils/Streami18n';
+    import type {
+        DefaultAttachmentType,
+        DefaultChannelType,
+        DefaultCommandType,
+        DefaultEventType,
+        DefaultMessageType,
+        DefaultReactionType,
+        DefaultUserType,
+        UnknownType,
+    } from '../../types/types';
+```
+
+### __Destructure Props__
+
+If more than __1__ prop is being used within a method, it should be passed in as `props` and then destructured on the first line of the body.
+
+_Example single prop:_
+
+```tsx
+export const EmptyStateIndicator: React.FC<EmptyStateProps> = ({
+  listType,
+}) => {
+  ...
+};
+```
+
+_Example multiple props:_
+
+```tsx
+export const Avatar: React.FC<AvatarProps> = (props) => {
+  const {
+    containerStyle,
+    image: imageProp,
+    imageStyle,
+    name,
+    online,
+    presenceIndicator: presenceIndicatorProp,
+    presenceIndicatorContainerStyle,
+    size,
+    testID,
+  } = props;
+  ...
+};
+```
+
+### __Variable Declaration__
+
+Do not use `var`. Use `const` and avoid using `let` unless a creative solution cannot be implemented.
+
+Bad:
+
+```tsx
+let extra = attachment?.actions?.length ? 'actions' : '';
+if (
+    componentType === 'card' &&
+    !attachment.image_url &&
+    !attachment.thumb_url
+) {
+    extra = 'no-image';
+}
+```
+
+Good:
+
+```tsx
+const extra =
+    componentType === 'card' && !attachment?.image_url && !attachment?.thumb_url
+      ? 'no-image'
+      : attachment?.actions?.length
+      ? 'actions'
+      : '';
+```
```

Comment:
I personally try to stay away from nested ternary statements. I would lean towards using your 'bad' solution in this case. Open to hear opinions on this of course.

Reply:
For small statements ternary is fine. But if the condition is big enough, I would switch to non-ternary way.

Reply:
const with TypeScript:
```
const hello = "Hello" as string | undefined
if (hello !== undefined) {
    ["A", "B"].forEach(
        name => console.log(`${hello.toUpperCase()}, ${name}`) // OK
    )
}
```
With let, in strict mode, TypeScript detects an error:
```
let hello = // …
// …
        name => console.log(`${hello.toUpperCase()}, ${name}`)
//                             ^__ error here: Object is possibly 'undefined'.
```

Reply:
Yes thats fine. I just think its case dependent, and should not be enforced by this doc.

Reply:
if there's a variable you'd want to use `let` instead of `const` for because it would have a lot of ternaries then it's best to create a function to return that variable value and within the function have individual returns for each of the situations. this helps with creating code for separate purposes as well as maintaining typescript value